src/Razor/src/Microsoft.CodeAnalysis.Razor.Workspaces/GoToDefinition/AbstractRazorComponentDefinitionService.cs

```diff
@@ -87,4 +88,6 @@ private async Task<LspRange> GetNavigateRangeAsync(IDocumentSnapshot documentSna
         // at least then press F7 to go there.
         return VsLspFactory.DefaultRange;
     }
+
+    protected abstract ValueTask<SyntaxTree> GetCSharpSyntaxTreeAsync(IDocumentSnapshot documentSnapshot, RazorCodeDocument codeDocument, CancellationToken cancellationToken);
```

Comment:
Hmmm... it feels a bit odd to pass an `IDocumentSnapshot` _and_ a `RazorCodeDocument`. If the `RazorCodeDocument` comes from the `IDocumentSnapshot`, I would expect all overrides to implement in terms of either the `IDocumentSnapshot` _or_ the `RazorCodeDocument`. Also, there's no guarantee that the objects passed in are related. For example, it would completely OK to pass in a `RazorCodeDocument` that does not belong to the `IDocumentSnapshot` that's passed in. So, this smells a bit funny to me.

Perhaps `IDocumentSnapshot` should just have a `GetCSharpSyntaxTreeAsync(...)` method?

Reply:
I can just get the code document from the snapshot. I didn't do that originally because this method wasn't async, but now that it is, that seems like the easiest solution.